### PR TITLE
fix: Apply icon color correctly inside native AppBarButton

### LINKED ISSUE #6732: [iOS][Android] Cannot change the color of the AppBarButtons on a native commandbar.
## Current behavior
Using the NativeCommandBar, the color defined in the `Foreground` Property of a BitmapIcon, nested inside an `AppBarButton`, doesn't appear on iOS and Android. 

The documention mentions that only a `SolidColorBrush` is supported for the color of the button, which is what I used in the sample below. 

## Expected behavior
Defining a color for an `AppBarButton`, the color should be seen on mobile devices.

## Screenshots 
On Android:
image
On iOS
image

On UWP
image

## How to reproduce it (as minimally and precisely as possible)
- Launch the app
- Notice how the cog icon on the CommandBar is gray, but is defined as blue in the Xaml.

on UWP, which uses the `XamlDefaultCommandBar` mode, the color is visible on the commandBar.

[CommandBarColoredCommands.zip](https://github.com/unoplatform/uno/files/6971813/CommandBarColoredCommands.zip)

## Works on UWP/WinUI

Yes, but it doesn't use the same mode as the commandbar on mobile (`XamlDefaultCommandBar` vs `NativeDefaultCommandBar`). 

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [x] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):
Uno.Core: 2.4.0
Uno.UI: 3.9.7
Uno.UI.RemoteControl: 3.9.7

Affected platform(s):

- [x] iOS
- [x] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: 16.10.3)
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [x] Resharper

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): #6732 (Android only is fixed)

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Icons inside AppBarButton don't get their color correctly.

## What is the new behavior?

Correctly set their color.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Controls/AppBarButtonRenderer.Android.cs [hunk 1]
```diff
@@ -152,14 +152,14 @@ protected override void Render()
 			native.SetVisible(element.Visibility == Visibility.Visible);
 
 			// Foreground
-			var foreground = element.Foreground as SolidColorBrush;
-			var foregroundColor = foreground?.Color;
+			var foreground = (element.Icon?.Foreground ?? element.Foreground) as SolidColorBrush;
```
Comment (Youssef1313) [inline src/Uno.UI/Controls/AppBarButtonRenderer.Android.cs:155]: I'm not sure if it's correct to fallback to `element.Foreground` here.

**Review by ajpinedam — APPROVED** [review]

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs [hunk 2]
```diff
@@ -30,6 +30,27 @@ public void NativeCommandBar_Automated()
...
+			var bitmap = screenshot.GetBitmap();
+			for (int x = (int)rect.Right; x > rect.CenterX; x--)
+			{
+				if (bitmap.GetPixel(x, (int)rect.CenterY) is { A: 255, R: 255, G: 0, B: 0 })
+				{
+					return;
+				}
+			}
+
+			Assert.Fail("Expected a red pixel.");
```
Comment (Youssef1313) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs:51]: I absolutely don't like this approach. But I have tried hard to get it working in a better way and couldn't 😕

Comment (MartinZikmund) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs:51]: Can the `AppBarButton` be aligned to the right horizontally? Then you could just sample some pixel with a static offset from the right. But I am not sure what are the paddings applied to the native command bars.

File: src/SamplesApp/SamplesApp.UITests/Constants.cs [hunk 3]
```diff
@@ -15,6 +15,6 @@ public partial class Constants
 		public const string iOSDeviceNameOrId = "iPad Pro (12.9-inch) (4th generation)";
 
 		// Default active platform when running under Visual Studio test runner
-		public const Platform CurrentPlatform = Platform.Browser;
+		public const Platform CurrentPlatform = Platform.Android;
```
Comment (MartinZikmund) [inline src/SamplesApp/SamplesApp.UITests/Constants.cs:18]: Revert this change please

**Review by MartinZikmund — APPROVED** [review]

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs (hunk 2 above)
Comment (Youssef1313) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs:51]: If you mean to eliminate the loop, I can do that without changing any alignments. But I'm not sure using a static offset is going to be stable for different screen sizes and resolutions.

Comment (jeromelaban) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs:51]: A static offset is likely not going to work... Let's keep this approach for now, even if it's a blunt.

**Comment by Youssef1313** [comment]
@MartinZikmund Can you re-run the failing tests? (I don't have permissions to do so)

**Comment by jeromelaban** [comment]
The UI test did not work it seems:
image

For wasm it's expected as it's a native command bar test, which may not apply, but for iOS, the test should be working though.

https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=1563684&_a=resultSummary&resultId=100000

Looking at the screenshots, the result does not seem to be correct but this is expected as the XAML is supposed to be running on Android only:

https://github.com/unoplatform/uno/blob/cf14a756053d9c1314cd50c868b0ea6f42d9ac38/src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/CommandBar/CommandBar_Native_With_Content.xaml#L53

You'll certainly need to create a new sample XAML file to test your scenario all available platforms.

**Comment by Youssef1313** [comment]
@jeromelaban This doesn't seem to be working on iOS. I don't have an iPhone or a Mac to try figuring out what's happening. Can we skip the test for iOS for now and open a new issue tracking that?

**Comment by jeromelaban** [comment]
You'll need to exclude this test for WebAssembly (the CommandBar is not set for this platform in the test). As for iOS, you can troubleshoot by looking at the screenshots there: https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=1564386&_a=resultSummary&resultId=100000

**Comment by Youssef1313** [comment]
Yeah I saw the screenshot in CI. For iOS, the icon doesn't seem to appear at all. Hard to investigate without a real device to debug. I reverted the iOS change temporarily to see if the icon will start appearing, after that let's see how we can move forward.

**Comment by Youssef1313** [comment]
@jeromelaban I skipped the test for iOS for now since I'm not sure what's going on.
